d2/src/main/java/com/linkedin/d2/balancer/simple/UriLoadBalancerSubscriber.java

```diff
@@ -75,7 +76,16 @@ protected void handlePut(final String cluster, final UriProperties uriProperties
           {
             Map<Integer, PartitionData> partitionDataMap = uriProperties.getPartitionDataMap(uri);
             TrackerClient client = trackerClients.get(uri);
-            if (client == null || !client.getPartitionDataMap().equals(partitionDataMap))
+
+            Optional<Map<String, Object>> newUriSpecificProperties = Optional.ofNullable(uriProperties.getUriSpecificProperties())
+              .map(uriSpecificProperties -> uriSpecificProperties.get(uri));
+
+            Optional<Map<String, Object>> oldUriSpecificProperties = Optional.ofNullable(_simpleLoadBalancerState.getUriProperties(clusterName))
+              .map(LoadBalancerStateItem::getProperty)
+              .map(UriProperties::getUriSpecificProperties)
+              .map(uriSpecificProperties -> uriSpecificProperties.get(uri));
+
+            if (client == null || !client.getPartitionDataMap().equals(partitionDataMap) || !newUriSpecificProperties.equals(oldUriSpecificProperties))
```

Comment:
Do we need to add this additional check `!newUriSpecificProperties.equals(oldUriSpecificProperties)`? I think adding uri specific properties will involve marking down and marking up nodes anyways.

Reply:
I'm not sure why but when testing locally the client did not receive the markDown update, only markUp. I'd need to ask some one more familiar with the zookeeper internals to understand why this is happening. But I can reproduce it and confirm the node is actually marked down/up in zk by looking at cversion. So looks like the check is needed.